deploy/definitions/main.template.json

```diff
@@ -0,0 +1,239 @@
+                    "keySource": "Microsoft.Storage"
+                },
+                "accessTier": "Hot"
+            }
+        },
+        {
+            "type": "Microsoft.Web/serverfarms",
+            "kind": "functionapp",
+            "name": "[parameters('functionAppName')]",
+            "apiVersion": "2015-08-01",
+            "location": "[resourceGroup().location]",
+            "sku": {
+                "name": "Y1",
+                "tier": "Dynamic",
+                "size": "Y1",
+                "family": "Y",
+                "capacity": 0
+            },
+            "properties": {
+                "name": "[parameters('functionAppName')]"
+            },
+            "dependsOn": []
+        },
+        {
+            "apiVersion": "2015-08-01",
+            "type": "Microsoft.Web/sites",
+            "name": "[parameters('functionAppName')]",
+            "location": "[resourceGroup().location]",
+            "kind": "functionapp",
+            "dependsOn": [
+                "[resourceId('Microsoft.Web/serverfarms', parameters('functionAppName'))]",
+                "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]"
+            ],
+            "properties": {
+                "name": "[parameters('functionAppName')]",
+                "serverFarmId": "[resourceId('Microsoft.Web/serverfarms',parameters('functionAppName'))]"
+            },
+            "resources": [
+                {
+                    "apiVersion": "2015-08-01",
+                    "name": "swap",
+                    "type": "slots",
+                    "tags": {
+                        "displayName": "[concat('swap', ' Deployment Slot')]"
+                    },
+                    "location": "[resourceGroup().location]",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Web/Sites', parameters('functionAppName'))]"
+                    ],
+                    "properties": {},
+                    "resources": [
+                        {
+                            "apiVersion": "2015-08-01",
+                            "type": "config",
+                            "name": "web",
+                            "dependsOn": [
+                                "[resourceId('Microsoft.Web/sites', parameters('functionAppName'))]",
+                                "[resourceId('Microsoft.Web/sites/slots', parameters('functionAppName'), 'swap')]"
+                            ],
+                            "properties": {
+                                "use32BitWorkerProcess": false,
+                                "cors": {
+                                    "allowedOrigins": [
+                                        "*"
+                                    ]
+                                }
+                            }
+                        }
+                    ]
+                },
+                {
+                    "apiVersion": "2015-08-01",
```

Comment:
Why do we have this as a separate resource? We can specify these properties above, right? When the Function App is being deployed?